docs/core/local-development.mdx

```diff
@@ -0,0 +1,349 @@
+---
+title: "Local Development"
+description: "Develop and test cloud applications entirely on your local machine"
+---
+
+Suga's local development environment (`suga dev`) lets you build and test cloud applications entirely on your local machine without cloud access, accounts, or costs. This page explains how local development works, what's emulated, and best practices for productive local workflows.
+
+## The Development Server
+
+The `suga dev` command starts a local development server that:
+
+- **Emulates cloud resources** - Buckets, entrypoints, databases
+- **Runs your services** - Using the `dev.script` from your `suga.yaml`
+- **Hot reloads code** - Automatically restarts services when files change
+- **Provides local URLs** - Access your application at `localhost` ports
+
+```bash title="Start local development"
+suga dev
+```
+
+```bash title="Example output"
+⚡ Suga v0.0.1
+   - App: my-app
+   - Addr: :50051
+
+Services
+
+✓ Starting [api]
+
+Entrypoints
+
+✓ Starting [web] http://localhost:3000
+
+Use Ctrl-C to exit
+```
+
+## What Gets Emulated
+
+### Services
+
+Your application services run locally using their `dev.script`:
+
+```yaml title="suga.yaml"
+services:
+  api:
+    dev:
+      script: npm run dev
+  worker:
+    dev:
+      script: python main.py
+```
+
+Each service:
+- Runs as a separate process
+- Gets automatically restarted on code changes
+- Receives environment variables from `suga.yaml`
+- Can communicate with emulated resources
+
+### Buckets (Object Storage)
+
+Buckets are emulated using the local filesystem:
+
+```
+project-root/
+└── .suga/
+    └── buckets/
+        └── uploads/        # Bucket named "uploads"
+            ├── file1.txt
+            └── file2.jpg
+```
+
+**Features:**
+- Read/write/delete operations work like cloud storage
+- Files persist between `suga dev` sessions
+- Pre-seed buckets by placing files in `.suga/buckets/{bucket-name}/`
+- View files directly in the file explorer
+
+**SDK Operations:**
+
+Use the generated Suga client to interact with buckets:
+
+```typescript title="Works locally and in production"
+// Write file
+await suga.uploads.write('file.txt', Buffer.from('Hello!'));
+
+// Read file
+const data = await suga.uploads.read('file.txt');
+
+// Delete file
+await suga.uploads.delete('file.txt');
+```
+
+Local behavior:
+- `write()` creates files in `.suga/buckets/uploads/`
+- `read()` reads from `.suga/buckets/uploads/`
+- `delete()` removes files from `.suga/buckets/uploads/`
+
+<Card title="Suga Client Libraries" icon="code" href="/develop/suga-client" horizontal>
+  Learn about generated client libraries for cloud-agnostic resource access
+</Card>
+
+### Entrypoints (HTTP Routing)
+
+Entrypoints provide local HTTP servers with routing:
+
+```yaml title="suga.yaml"
+entrypoints:
```

Comment:
Maybe its good to show what its connecting to.
````suggestion
```yaml title="suga.yaml"
services:
  api: ...
buckets:
  frontend: ...
entrypoints:
````